common/src/main/java/org/opensearch/ml/common/MLModelGroup.java

```diff
@@ -0,0 +1,206 @@
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
+@Getter
+public class MLModelGroup implements ToXContentObject {
+    public static final String MODEL_GROUP_NAME_FIELD = "name"; //name of the model group
+    // We use int type for version in first release 1.3. In 2.4, we changed to
+    // use String type for version. Keep this old version field for old models.
+    public static final String DESCRIPTION_FIELD = "description"; //description of the model group
+    public static final String LATEST_VERSION_FIELD = "latest_version"; //latest model version added to the model group
+    public static final String BACKEND_ROLES_FIELD = "backend_roles"; //back_end roles as specified by the owner/admin
+    public static final String OWNER = "owner"; //user who creates/owns the model group
+
+    public static final String ACCESS = "access"; //assigned to public, private, or null when model group created
+    public static final String MODEL_GROUP_ID_FIELD = "model_group_id"; //unique ID assigned to each model group
+    public static final String CREATED_TIME_FIELD = "created_time"; //model group created time stamp
+    public static final String LAST_UPDATED_TIME_FIELD = "last_updated_time"; //updated whenever a new model version is created
+    //SHA256 hash value of model content.
+
+    @Setter
+    private String name;
+    private String description;
+    private int latestVersion;
+    private List<String> backendRoles;
+    private User owner;
+
+    private String access;
+
+    private String modelGroupId;
+
+    private Instant createdTime;
+    private Instant lastUpdatedTime;
+
+
+    @Builder(toBuilder = true)
+    public MLModelGroup(String name, String description, int latestVersion,
+                        List<String> backendRoles, User owner, String access,
+                        String modelGroupId,
+                        Instant createdTime,
+                        Instant lastUpdatedTime) {
+        this.name = name;
+        this.description = description;
+        this.latestVersion = latestVersion;
+        this.backendRoles = backendRoles;
+        this.owner = owner;
+        this.access = access;
+        this.modelGroupId = modelGroupId;
+        this.createdTime = createdTime;
+        this.lastUpdatedTime = lastUpdatedTime;
+    }
+
+
+    public MLModelGroup(StreamInput input) throws IOException{
+        name = input.readString();
+        description = input.readOptionalString();
+        latestVersion = input.readInt();
+        backendRoles = input.readOptionalStringList();
+        if (input.readBoolean()) {
+            this.owner = new User(input);
+        } else {
+            this.owner = null;
+        }
+        access = input.readOptionalString();
+        modelGroupId = input.readOptionalString();
+        createdTime = input.readOptionalInstant();
```

Comment:
Not a bug, but is  `access`, `createdTime` and `lastUpdatedTime` are optional?

Reply:
access was optional previously. Its mandatory now. And the times are optional in model. So just used a similar type here. We can actually make them all mandatory later